src/compiler/build/compiler-ctx.ts

```diff
@@ -103,6 +103,8 @@ export const getModuleLegacy = (_config: d.Config, compilerCtx: d.CompilerCtx, s
       potentialCmpRefs: [],
       staticSourceFile: null,
       staticSourceFileText: '',
+      sourceMapPath: null,
```

Comment:
AFAIK, this method helps build the moduleMap, I don't understand yet why it's named with "legacy" yet, but generally this should help to build the object that will ultimately be a part of the moduleMap. 

Interestingly enough, this is in some way already on the buildCtx too, since buildCtx takes in the compilerCtx and uses that to do some internal stuff throughout the life of a build. So, this is technically available on the buildCtx, and may in fact be used by it too, but I don't yet know if it is for sure. At a cursory search through the codebase, it doesn't seem to be used. (I searched with .compilerCtx.moduleMap)

I assume it's set up this way to save on memory. If the module map was instantiated and populated for each build, there would be a lot of processing power and disk reads that would be prohibitive. So having it at a higher level cache and providing it down to the buildCtx should help mitigate that. 

I swear I'll get around to an ADR on compilerCtx and buildCtx lol. There's a lot to unravel about these two concepts.